### DISCUSSION TITLE
Support `markdown` package

### DISCUSSION THREAD

**Discussion by Neved4** [tectonic-typesetting/tectonic#1024] [Q&A]
@pkgw Lately I've been working with sources that use the `markdown` package, and I wonder whether we consider reopening this https://github.com/tectonic-typesetting/tectonic-staging/issues/6. As of `tectonic 0.12.0` it's not working.

The package depends on a `texlua` call in combination with `shell-escape` to work properly, but with XeTeX it works without issue:
```console
$ latexmk -pdfxe -outdir=. -shell-escape test.tex
```

With `tectonic -Z shell-escape test.tex` it seems to compile without errors, but on further inspection the rendered PDF is not correct, and checking `test.markdown.lua` we see the offending bit
```lua
[...]

 print('\\markdownError{An error was encountered while executing Lua code}
     {For further clues, examine the file "./test.markdown.err"}')
```

Note that in the [docs](https://osl.ugr.es/CTAN/macros/generic/markdown/markdown.pdf), redefining `\markdownOptionOutputDir` may be necessary to match the output directory when it's a different value than the default.

MWE `test.tex`:
```latex
\documentclass{article}
  \usepackage{markdown}

\begin{document}
  \begin{markdown}
  # Grocery list

  ## Food
  - baked beans
  - spaghetti

  ## Stationery
  - writing pad
  - pencils
  \end{markdown}
\end{document}
```

**Comment by vlasakm** [comment]
Ad reopening the issue: Until somebody can use the `markdown` package succesfully I agree with reopening the issue.

Ad the issue: The `markdown` package probably requires `texlua` -- i.e. `luatex` in disguise as a `lua` interpreter with a few libraries on top.

I don't know whether it uses some LuaTeX specific API, but if it doesn't, Tectonic could just provide the "`lua`" part of LuaTeX, which could also solve problems with other packages, which just need to run some Lua code, but don't necessarily need the `lua` from `luatex`, e.g. https://github.com/tectonic-typesetting/tectonic/issues/838.

A bit off-topic, but justifying the LuaTeX inclusion: It would also be nice to include LuaTeX in full:

 - It already solved a lot of the problems Tectonic is trying to solve by being based on and extending XeTeX (Unicode, OpenType fonts, readable C code, ...)
 - Doesn't have any real "external system library" dependencies like HarfBuzz (though it's more complicated nowadays), TecKit or font-config. (That has its own problems as well, but it seems to work for people well.)
 - Generates PDFs directly (no DVI intermediate form posprocessed with `dvipdfmx` which has a lot of issues and is another thing Tectonic needs to maintain).
 - Has full support for microtype.
 - Both of the above features are due to it being descendant of pdfTeX, thus a lot of the pdfTeX compatibility issues would also get solved nicely by that (e.g. `\pdfoutput=1`).
